test/mpi/update_gitignore_binaries.py

```diff
@@ -0,0 +1,46 @@
+#This file will automatically add newly found binaries in subdirectories to
+#the .gitignore file. This must be run from mpich/test/mpi for proper usage
+import subprocess
+
+#CLI command to get all binaries
+ignored_binaries = ('-not -name "*.*" -not -name "README" -not -name "Makefile" -not -name "testlist"'
+        ' -not -name "ToDo" -not -name "checktests" -not -name "configure" -not -name "libtool" -not -name "runtests"')
+
```

Comment:
We only need to update `.gitignore` every once in a while, so I am not sure it is worth it to maintain a separate script for it. Consider that keeping the `ignored_binaries` always updated is as challenging as keeping `.gitignore` updated itself. I suggest remove this script from the PR.